src/global-lexical.lisp

```diff
@@ -45,10 +45,9 @@
 (defun (setf top-level-binding-value) (new-value binding)
   (setf (car binding) new-value))
 
-(defmacro define-global-lexical (var val)
+(defmacro define-global-lexical (var type)
```

Comment:
Can we rename this to DEFINE-GLOBAL-COATON-VALUE or something along those lines?

I also think it needs documentation to establish what needs to be true about VAR when this macro is expanded. (How does VAR get a value? When will it get a value?)

The top level definitions are sufficiently tricky, and have evolved a lot since Coalton 0.000000001, so I think some docs would be good.